akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/ReplicatedShardingDirectReplicationSpec.scala

```diff
@@ -49,6 +52,55 @@ class ReplicatedShardingDirectReplicationSpec extends ScalaTestWithActorTestKit
       replicaAProbe.expectNoMessage() // no publishing to the replica emitting it
     }
 
+    "not forward messages for a different type name" in {
+      val replicaAProbe = createTestProbe[ShardingEnvelope[PublishedEvent]]()
+
+      val replicationActor = spawn(
+        ShardingDirectReplication(
+          "ReplicatedShardingSpec",
+          None,
+          replicaShardingProxies = Map(ReplicaId("ReplicaA") -> replicaAProbe.ref)))
+
+      val upProbe = createTestProbe[Done]()
+      replicationActor ! ShardingDirectReplication.VerifyStarted(upProbe.ref)
+      upProbe.receiveMessage() // not bullet proof wrt to subscription being complete but good enough
+
+      val event = PublishedEventImpl(
+        ReplicationId("ADifferentReplicationId", "pid", ReplicaId("ReplicaA")).persistenceId,
+        1L,
+        "event",
+        System.currentTimeMillis(),
+        Some(new ReplicatedPublishedEventMetaData(ReplicaId("ReplicaA"), VersionVector.empty)))
+      system.eventStream ! EventStream.Publish(event)
+
+      replicaAProbe.expectNoMessage()
+    }
+
+    "ignore messages not from replicated event sourcing" in {
+      val replicaAProbe = createTestProbe[ShardingEnvelope[PublishedEvent]]()
+
+      val replicationActor = spawn(
+        ShardingDirectReplication(
+          "ReplicatedShardingSpec",
+          None,
+          replicaShardingProxies = Map(ReplicaId("ReplicaA") -> replicaAProbe.ref)))
+
+      val upProbe = createTestProbe[Done]()
+      replicationActor ! ShardingDirectReplication.VerifyStarted(upProbe.ref)
+      upProbe.receiveMessage() // not bullet proof wrt to subscription being complete but good enough
+
+      LoggingTestKit.error[IllegalArgumentException].withOccurrences(0).expect {
```

Comment:
`withOccurrences(0)`? so it would log error on receiving side if ShardingDirectReplication would have sent it? Could you add some more specific assert condition on the log message so we see what IllegalArgumentException this relates to?

Reply:
This was thrown before we caught the parsing of the persistence id as a replicated one so the actor was stopped. If we had a expectNotTerminated that would be better